flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/TableEnvImpl.scala

```diff
@@ -65,16 +58,20 @@ import org.apache.flink.table.typeutils.TimeIndicatorTypeInfo
 import org.apache.flink.table.validate.FunctionCatalog
 import org.apache.flink.types.Row
 
-import _root_.scala.annotation.varargs
 import _root_.scala.collection.JavaConverters._
 import _root_.scala.collection.mutable
 
 /**
-  * The abstract base class for batch and stream TableEnvironments.
+  * The abstract base class for the implementation of batch and stream TableEnvironments.
   *
   * @param config The configuration of the TableEnvironment
   */
-abstract class TableEnvironment(val config: TableConfig) {
+abstract class TableEnvImpl(val config: TableConfig) extends TableEnvironment {
+
+  // Init PlannerConfig with the DefaultPlannerConfig
+  if (config.getPlannerConfig == null) {
+    config.setPlannerConfig(DefaultPlannerConfig.createBuilder().build())
```

Comment:
```suggestion
    config.setPlannerConfig(DefaultPlannerConfig.DEFAULT)
```

use `DefaultPlannerConfig.DEFAULT` instead ?

Reply:
Let's not set anything here. `TableEnvImpl` is meant to be moved to api module. `DefaultPlannerConfig` won't be in the scope anymore.

This is where `Optional` comes in handy. On the planner side we will just do: 

```
plannerConfig.unwrap(DefaultPlannerConfig.class).orElse(DefaultPlannerConfig.DEFAULT)
```